src/native/tls_context_options.c

```diff
@@ -145,6 +156,76 @@ jlong JNICALL Java_software_amazon_awssdk_crt_io_TlsContextOptions_tlsContextOpt
             aws_jni_throw_runtime_exception(env, "aws_tls_ctx_options_init_client_mtls_with_pkcs11 failed");
             goto on_error;
         }
+    } else if (jni_custom_key_op) {
+
+        jobject jni_custom_key_op_handle = (*env)->GetObjectField(
+            env, jni_custom_key_op, tls_context_custom_key_operation_options_properties.operation_handler_field_id);
+        if (!jni_custom_key_op_handle) {
+            aws_jni_throw_runtime_exception(
+                env, "could not get custom operation handler from jni_custom_key_op_handle!");
+            goto on_error;
+        }
+
+        tls->custom_key_op_handler = aws_custom_key_op_handler_java_new(env, allocator, jni_custom_key_op_handle);
+
+        /* Certificate needs to be set, but there are multiple ways to get it */
+        jstring jni_custom_key_op_cert_path = (*env)->GetObjectField(
+            env, jni_custom_key_op, tls_context_custom_key_operation_options_properties.certificate_file_path_field_id);
+        jstring jni_custom_key_op_cert_contents = (*env)->GetObjectField(
+            env,
+            jni_custom_key_op,
+            tls_context_custom_key_operation_options_properties.certificate_file_contents_field_id);
+
+        if (jni_custom_key_op_cert_path && jni_custom_key_op_cert_contents) {
+            /* Cannot have both a certificate path and data - use one or the other */
+            aws_jni_throw_runtime_exception(
+                env, "Custom key operation handler: cannot have both certificate file path and certificate contents!");
+            goto on_error;
+        } else if (jni_custom_key_op_cert_contents) {
+            /* If we have the certificate contents, then use it directly */
+            tls->certificate = aws_jni_new_string_from_jstring(env, jni_custom_key_op_cert_contents);
+            struct aws_byte_cursor certificate_byte_cursor = aws_byte_cursor_from_string(tls->certificate);
```

Comment:
not checking result of aws_jni_new_string_from_jstring() here, but we do check it in the next if-block

figure out which approach is the right one